src/Evolution/Executables/Burgers/EvolveBurgers.hpp

```diff
@@ -158,7 +158,8 @@ struct EvolutionMetavars {
 
               Parallel::PhaseActions<
                   Phase, Phase::RegisterWithObserver,
-                  tmpl::list<observers::Actions::RegisterWithObservers<
+                  tmpl::list<Actions::AdvanceTime,
```

Comment:
Minor suggestion: Can you do this at the end of the inititalization phase instead of in the registering phase? Makes more sense semantically

Reply:
That's not the same place in the algorithm... Registration happens after time stepper initialization.

Reply:
Right, so this would be better placed at the end of `Phase::InitializeTimeStepperHistory` I think.

Reply:
I think it would make more sense in the time-stepper initialization, but moving it there is non-trivial.  The modifications to the TimeId would have to occur in the middle of the Cleanup action, so that action would have to be split.  This is not particularly difficult, but it is more than I want to get into in this PR.

Reply:
Is the issue that the time-stepper initialization phase is a loop and you want to invoke AdvanceTime only when it’s finished? In that case, how about you add a new phase for this like `Phase::PrepareEvolution`?

Reply:
we want to minimize the number of phases as much as possible

Reply:
Sure. I don’t think a phase like this would have any significant impact on runtime though. Anyway, perhaps a comment in the code could help readers understand the reasoning you described here. Please consider this suggestion non-blocking for merging this PR.